### PR TITLE
Updated packages, updated dockerfiles, added more utilities

### PR DESCRIPTION
**Pull request by 252afh**

### CODE REVIEW

File: {{cookiecutter.project_slug}}/docker/tests/entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+
+set -o errexit
+set -o nounset
+
+timeout 60s bash -c "until pg_isready -d $DATABASE_URL; do sleep 5; done"
```
Comment (Singletoned) [inline {{cookiecutter.project_slug}}/docker/tests/entrypoint.sh]: We should probably be using `wait-for-db` here

File: {{cookiecutter.project_slug}}/{{cookiecutter.package_name}}/jinja2.py [hunk 2]
```diff
@@ -1,6 +1,20 @@
...
+
+
+def url(path, *args, **kwargs):
+    assert not (args and kwargs)
+    return reverse(path, args=args, kwargs=kwargs)
+
+
+def markdown(text, cls=None):
+    html = markdown_converter.render(text).strip()
+    html = html.replace("<p>", f'<p class="{cls or ""}">', 1).replace("</p>", "", 1)
```
Comment (Singletoned) [inline {{cookiecutter.project_slug}}/{{cookiecutter.package_name}}/jinja2.py:27]: What is this supposed to be doing?  We probably need a docstring or comment in here to explain (or remove this line)

**Review by Singletoned — CHANGES_REQUESTED** [review]

**Comment by Singletoned** [comment]
I think we probably don't need the PaaS deploy and proxy files in there

File: {{cookiecutter.project_slug}}/.github/workflows/proxy.yml [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+          ROUTE_SERVICE_APP_NAME: ${{ matrix.app }}-proxy
+          ROUTE_SERVICE_NAME: ${{ matrix.app }}-route
+          PROTECTED_APP_NAME: ${{ matrix.app }}
+          ALLOWED_IPS: ${{ steps.get_ips.outputs.ips }}
+          HTPASSWD: ${{ secrets.HTPASSWD }}
+        working-directory: ./proxy
+        run: |
+            ./proxy-deploy.sh
+
+      - uses: actions/checkout@v2
```
Comment (nmenezes0) [inline {{cookiecutter.project_slug}}/.github/workflows/proxy.yml]: Let's get rid of the Slack Notifications - no-one uses them, and some of us don't understand them.

**Comment by nmenezes0** [comment]
Please can you add a README with instructions on how to generate a new cookiecutter project?

File: {{cookiecutter.project_slug}}/.github/workflows/checks.yml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+jobs:
+  check_web:
+    name: Check Python
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Set up Python 3.9
```
Comment (nmenezes0) [inline {{cookiecutter.project_slug}}/.github/workflows/checks.yml:24]: 3.9 or 3.8?
Maybe 3.9? (I think has to be <=3.9 for Notify?)

Comment (nmenezes0) [reply {{cookiecutter.project_slug}}/.github/workflows/checks.yml:24]: 3.8 definitely referenced elsewhere

**Comment by nmenezes0** [comment]
Got an error when trying to run:
```
Traceback (most recent call last):
  File "/Users/nina.menezes/.virtualenvs/testenv/bin/cookiecutter", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/Users/nina.menezes/.virtualenvs/testenv/lib/python3.11/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nina.menezes/.virtualenvs/testenv/lib/python3.11/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/Users/nina.menezes/.virtualenvs/testenv/lib/python3.11/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nina.menezes/.virtualenvs/testenv/lib/python3.11/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nina.menezes/.virtualenvs/testenv/lib/python3.11/site-packages/cookiecutter/cli.py", line 194, in main
    cookiecutter(
  File "/Users/nina.menezes/.virtualenvs/testenv/lib/python3.11/site-packages/cookiecutter/main.py", line 114, in cookiecutter
    result = generate_files(
             ^^^^^^^^^^^^^^^
  File "/Users/nina.menezes/.virtualenvs/testenv/lib/python3.11/site-packages/cookiecutter/generate.py", line 373, in generate_files
    generate_file(
  File "/Users/nina.menezes/.virtualenvs/testenv/lib/python3.11/site-packages/cookiecutter/generate.py", line 173, in generate_file
    tmpl = env.get_template(infile_fwd_slashes)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nina.menezes/.virtualenvs/testenv/lib/python3.11/site-packages/jinja2/environment.py", line 1010, in get_template
    return self._load_template(name, globals)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nina.menezes/.virtualenvs/testenv/lib/python3.11/site-packages/jinja2/environment.py", line 969, in _load_template
    template = self.loader.load(self, name, self.make_globals(globals))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nina.menezes/.virtualenvs/testenv/lib/python3.11/site-packages/jinja2/loaders.py", line 138, in load
    code = environment.compile(source, name, filename)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nina.menezes/.virtualenvs/testenv/lib/python3.11/site-packages/jinja2/environment.py", line 768, in compile
    self.handle_exception(source=source_hint)
  File "/Users/nina.menezes/.virtualenvs/testenv/lib/python3.11/site-packages/jinja2/environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File ".github/workflows/proxy.yml", line 89, in template
jinja2.exceptions.TemplateSyntaxError: unexpected char '&' at 2481
  File ".github/workflows/proxy.yml", line 89
    if: ${{ failure() && github.ref_name == 'prod' || failure() && github.ref_name == 'develop' }}
```

Might be ok once the `proxy` stuff is removed.

**Review by nmenezes0 — CHANGES_REQUESTED** [review]
Looks good overall, a couple of minor comments, I'd like to make sure I can get it to run before approving.

File: {{cookiecutter.project_slug}}/README.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+
+## Running tests
+
+    make test
+
+
+## Checking code
+
+    make check-python-code
```
Comment (nmenezes0) [inline {{cookiecutter.project_slug}}/README.md:17]: Add some stuff on using the cookiecutter, which as I recall was something like installing python package, cloning this repo, running some command like `cookiecutter cookie-cutter`.

Comment (nmenezes0) [reply {{cookiecutter.project_slug}}/README.md:17]: DOh, it was in the actual readme.
Just ignore me

**Review by nmenezes0 — APPROVED** [review]
All runs fine for me now - this is going to be really helpful for the next urgent project!

File: {{cookiecutter.project_slug}}/{{cookiecutter.package_name}}/jinja2.py [hunk 6]
```diff
@@ -1,6 +1,31 @@
 import jinja2
 from django.templatetags.static import static
 from django.urls import reverse
+from markdown_it import MarkdownIt
+
+markdown_converter = MarkdownIt()
```
Comment (nmenezes0) [inline {{cookiecutter.project_slug}}/{{cookiecutter.package_name}}/jinja2.py]: Change to `MarkdownIt("js-default")` to make it safe.
https://markdown-it-py.readthedocs.io/en/latest/security.html for background

**Review by nmenezes0 — CHANGES_REQUESTED** [review]
One comment on making markdown safe

**Review by nmenezes0 — APPROVED** [review]
Subject also to Ed's approval!

**Review by Singletoned — APPROVED** [review]

**Pull request merged**